File: backend/app/Models/Dispute.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Concerns\HasUuids;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class Dispute extends Model
{
    use HasFactory, HasUuids;
    protected $fillable = ['booking_id', 'reporter_id', 'type', 'description', 'evidence_urls', 'status', 'resolution', 'refund_amount', 'resolved_by', 'resolved_at'];
    protected function casts(): array { return ['evidence_urls' => 'array', 'refund_amount' => 'decimal:2', 'resolved_at' => 'datetime']; }
    public function booking(): BelongsTo { return $this->belongsTo(Booking::class); }
    public function reporter(): BelongsTo { return $this->belongsTo(User::class, 'reporter_id'); }
    public function resolver(): BelongsTo { return $this->belongsTo(User::class, 'resolved_by'); }
}

File: frontend/lib/core/models/dispute.dart
import '../utils/json_utils.dart';
import 'user.dart';

/// A dispute raised against a booking (matches `DisputeResource`).
class Dispute {
  final String id;
  final String bookingId;
  final String reporterId;
  final String type; // damage | late_return | no_show | fraud | other
  final String description;
  final List<String> evidenceUrls;
  final String status; // open | investigating | resolved | escalated | ...
  final String? resolution;
  final double? refundAmount;
  final DateTime? resolvedAt;
  final UserSummary? reporter;
  final UserSummary? resolver;
  final Map<String, dynamic>? booking;
  final DateTime? createdAt;

  const Dispute({
    required this.id,
    required this.bookingId,
    required this.reporterId,
    required this.type,
    required this.description,
    this.evidenceUrls = const [],
    required this.status,
    this.resolution,
    this.refundAmount,
    this.resolvedAt,
    this.reporter,
    this.resolver,
    this.booking,
    this.createdAt,
  });

  bool get isResolved => status == 'resolved';
  bool get isOpen => status == 'open' || status == 'investigating';

  factory Dispute.fromJson(Map<String, dynamic> json) => Dispute(
        id: asString(json['id']),
        bookingId: asString(json['booking_id']),
        reporterId: asString(json['reporter_id']),
        type: asString(json['type'], fallback: 'other'),
        description: asString(json['description']),
        evidenceUrls: asStringList(json['evidence_urls']),
        status: asString(json['status'], fallback: 'open'),
        resolution: asStringOrNull(json['resolution']),
        refundAmount: asDoubleOrNull(json['refund_amount']),
        resolvedAt: asDateTime(json['resolved_at']),
        reporter: json['reporter'] is Map
            ? UserSummary.fromJson(Map<String, dynamic>.from(json['reporter']))
            : null,
        resolver: json['resolver'] is Map
            ? UserSummary.fromJson(Map<String, dynamic>.from(json['resolver']))
            : null,
        booking: asMap(json['booking']),
        createdAt: asDateTime(json['created_at']),
      );
}
